### DISCUSSION TITLE
Creating a schema with minified version of a type

### DISCUSSION THREAD

**Discussion by aegirorn** [prisma/prisma#11877] [Q&A] [answered]
How can I create a Prisma schema for mongoDb that contains sort of a minified version of another object?

To explain better the data I am working with could look something like this:

```
const basicUser = {
  id: "62091d9ae17cad32fbe0eda5",
  DisplayName: "John Smith",
};
const post = {
  title: "My Title",
  content: "My content",
  author: basicUser,
};

const User = {
    id: "62091d9ae17cad32fbe0eda5",
    FirstName: "John",
    LastName: "Smith",
    DisplayName: "John Smith",
    EmailAddress:"john@test.com"
}
```

Thanks!

**Comment by nurul3101** [comment, marked as answer]
Hey @aegirorn 👋🏻 ,
We just released support for composite types (Known as Embedded Documents in MongoDB) for embedding documents inside documents.
From your use case, it seems that composite types should be helpful to you.
You can define `basicUser` as a type and reference it in your `post` model.
```prisma
type BasicUser {
  id          String
  displayName String
}

model Post {
  title   String
  content String
  author  BasicUser
}
```
Reference for using [composite types](https://www.prisma.io/docs/concepts/components/prisma-client/composite-types).
Please note that this feature is in preview and you need to be on version `3.10.0` or later.
Feel free to reach out in case you have any other questions.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
